### PR TITLE
Logs: Add `Download logs` button to log-browser

### LINKED ISSUE #50188: Loki Datasource: Make the "Downloads" button more prominent
**Problem**
Today figuring out how to download logs from the Explore view of a Loki datasource is difficult. 
Screen Shot 2022-06-03 at 12 12 05 PM

First the user has to (1) click on 'Inspector', then they have to (2) click on 'Data', and then they can (3) click one of two buttons (download logs or download csv). 

This makes discoverability challenging for what is a pretty common use case: downloading your logs as a file so you can use them in some other context. For example, some users want to download logs to attach them to a bug report. Others want to download them because they'd like to manipulate or search them in some other tool (e.g., one of the self serve users that we recently spoke to wanted to load his into a jupyter notebook for further processing). 

**Proposal**
We'd like to put the download button on the main screen to make it more discoverable. Furthermore, we'd like to have a single button which then gives you a choice of what format to download the logs in. Right now, we support downloading as a csv or as a txt file. In the future, we might support other formats. 

Here's a mockup of what it would look like after this change:
Screen Shot 2022-06-03 at 12 17 28 PM

For full design, we have a Figma link as well:
https://www.figma.com/file/7ZHtMoaK4RxBz3b6m6U0cW/Show-more-logs-%2B-Sorting?node-id=1709%3A9228

### PR DESCRIPTION
**Pull request by svennergr**
**WIP**:
This is still work in progress as it
- needs tests
- should not be merged until Grafana 9.2.0

**What this PR does / why we need it**:
In https://github.com/grafana/grafana/issues/50188 we decided to add a new `Downloads` button to the logs browser. This browser supports only txt-format for now.

**Which issue(s) this PR fixes**:

Fixes #50188

**Preview of the feature**:

https://user-images.githubusercontent.com/8092184/190149342-29e978fd-7a33-4945-94a0-5e536375f4c8.mov

### CODE REVIEW

File: public/app/features/inspector/InspectDataTab.tsx [hunk 1]
```diff
@@ -99,43 +96,19 @@ export class InspectDataTab extends PureComponent<Props, State> {
...
   };
 
   exportLogsAsTxt = () => {
     const { data, panel, app } = this.props;
+
     reportInteraction('grafana_logs_download_logs_clicked', {
       app,
       format: 'logs',
     });
-    const logsModel = dataFrameToLogsModel(data || [], undefined);
```
Comment (matyax) [inline public/app/features/inspector/InspectDataTab.tsx:119]: Great cleanup.

**Review by matyax — APPROVED** [review]
Looks great!

File: package.json [hunk 2]
```diff
@@ -173,6 +173,7 @@
     "babel-loader": "8.2.5",
     "babel-plugin-angularjs-annotate": "0.10.0",
     "babel-plugin-macros": "3.1.0",
+    "blob-polyfill": "7.0.20220408",
```
Comment (svennergr) [inline package.json:176]: I added that in order to add methods to the `Blob` object, which seem to be missing on JSDom.

Ref https://github.com/jsdom/jsdom/issues/2555

File: public/app/features/inspector/utils/download.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  saveAs(blob, fileName);
+}
+
+/**
+ * Downloads any object as JSON file.
+ *
+ * @param {*} json
+ * @param {string} title
+ */
+export function downloadAsJson(json: any, title: string) {
```
Comment (joshhunt) [inline public/app/features/inspector/utils/download.ts:72]: Don't use `any`. This type should not be in our vocabulary!

[`unknown` should work](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAEzgdzAGzgQ2QQQGcApQhACgCsywAuRcAazHTAEpEBvAKEUQCcAplBD8kxAMoB5AHIA6QlH4wwAcxjAAnlRpsA3NwC+3bqgzY8RUhQBEAC0GZsAGkRo4-TMhv7TrCwQkNOScDk5w9Dbunt6GvmZYuIHWYOQA2vaO2DauUR5eNgC6+kA)

Comment (svennergr) [reply public/app/features/inspector/utils/download.ts]: Thanks Josh! 🙏

**Pull request merged**